signingscript/src/signingscript/sign.py

```diff
@@ -286,13 +286,14 @@ def get_autograph_config(
     """
     file_base, file_extension = os.path.splitext(orig_path)
 
-    if not file_extension == ".xpi":
+    if file_extension not in (".xpi", ".zip"):
         raise SigningScriptError("Expected a .xpi")
 
-    id = _langpack_id(orig_path)
+    ext_id = _extension_id(orig_path, fmt)
     log.info("Identified {} as extension id: {}".format(orig_path, id))
+    kwargs = {'extension_id': ext_id}
     # Sign the appropriate inner files
-    await sign_file_with_autograph(context, orig_path, fmt, extension_id=id)
+    await sign_file_with_autograph(context, orig_path, fmt, **kwargs)
```

Comment:
I think you'll need to grab the keyid from somewhere before calling sign_file_with_autograph? Or maybe modify sign_with_autograph to get the keyid from the server object if the keyid value passed into the function is None?

Reply:
Ah, I thought that was already hooked up. Will do

Reply:
Done. While getting 100% coverage I found some bugs; fixed.